README.md
# OnCloudTime





release-notes.md
# Release Notes
------------------------------------------------------------------------------------------------------------------------
## 0.7.0 - Cognito Integration 

1. Add Cognito access to `ecsInstanceRole` and `ecsTaskExecutionRole` roles by attaching the `AmazonCognitoPowerUser` and `AmazonECSTaskExecutionRolePolicy` policy
2. Reset the credentials by running `sh docker_clear_secret_oncloudtime.sh`

## 0.5.8 - 17 Jan 19

1. Added docker deployment infrastructure for EC2 

2. Timesheets:
- Ability to create projects and project assignments 
- Projects can be either Agile or Waterfall 
- Users can submit timesheets for the current week 

## 0.5.6 -  14 Dec 2018
1. File paths for users 
    * Added a home folder within the user directories paths
2. File Uploads
  - Removed the convert to ORC and Parquet actions which are not relevant with the custom formats on the Templates 
3. Templates
    * Changed the options for format to Text, Parquet, Orc, JSON
    * Changed the delimiter to only a single value, comma 
    * When a user template is created for ORC, Parquet and JSON - external tables for the formats are created in addition to a text table 
    * When a core template is created for ORC, Parquet and JSON - external tables for the shared formats are created in addition to a text table 
    * Added a spinner when uploading a sample row 
4. Queries
    * Query output url uses the CSV file generated by athena when running te query 
5. File uploads
    * A file validation has been added to ensure that a file cannot be uploaded to a template more than once 
    * Only files uploaded to templates of type core can be shared as they have a shared directory across users 
    * Files are uploaded to the new directory structure which includes a folder for the template format 
6. Added a spinner showing progress for each of the forms on the site to show progress and prevent double submission
7. Reports - removed tables until the way to search columnar formats for uploaded files is resolved 

Deployment instructions:
=========================    
1. Unzip package 
2. Copy files 
3. Run deploy.sh to set permissions & clear cache `sh /var/www/html/oncloudtime/deploy.sh` 
4. Default credentials are:
    * Super Admin -  username: admin3@oncloudtime.com, password: 12345678
    * Admin - username: ATS003@oncloudtime.com, password: 12345678
    * Regular user - username: ATS0043@oncloudtime.com, password: 12345678
5. Documentation
    * User Manual - docs/oncloudtime-usermanual.pdf
------------------------------------------------------------------------------------------------------------------------
## 0.5.3 - 22 Oct 2018 
1. User Accounts
    * Fixed error on updating user account 
    * Added a validation to ensure a role is selected when creating or updating a user 
    * Merged all user account fields from the employee table 
2. SQS
    * All data used to execute any action is sent as part of the SQS message 
3. Templates 
    * Added handling for errors when loading the sample row 
4. Test Cases and flows documented in test-cases.md file in this directory 
5. NOT Done: 
    * Disabling the submit buttons after submission - requires changes in very many places so put it on hold  

Deployment instructions:
=========================    
1. Unzip package 
2. Copy files 
3. Run deploy.sh to set permissions & clear cache `sh /var/www/html/oncloudtime/deploy.sh` 
4. Run the script in sql/0.5.3.sql
5. Default credentials are:
    * Super Admin -  username: admin3@oncloudtime.com, password: 12345678
    * Admin - username: ATS003@oncloudtime.com, password: 12345678
    * Regular user - username: ATS0043@oncloudtime.com, password: 12345678

------------------------------------------------------------------------------------------------------------------------

## 0.5.2 - 10 Oct 2018 
1. Accounts
    * Fixed SNS topics not being created with special characters 
    * Remove all characters that are not numbers or leters from the S3 folders derived from account names as this was causing issues for athena output
2. User Accounts 
    * Users can now only login using their email addresses not user accounts 
    * Removed the concept of username, now internal as the uuid of the user account for backwards compatibility 
    * 
3. Security 
    * Changed session timeout to 18 hours instead of 20  minutes 
4. File Upload 
    * Hide the folder paths 
5. Install script 
    * Migrate to using shell backend to create the .aws_config file, and save the configurations 
    * Set needs_update to 'yes' in the install.yaml to unlock the installer
    * Add SNS topic option to the installer 


Deployment instructions:
=========================    
1. Unzip package 
2. Copy files 
3. Run deploy.sh to set permissions & clear cache `sh /var/www/html/oncloudtime/deploy.sh` 
4. Run the script in sql/0.5.2.sql
5. Run installer at https://oncloudtime.com/install 
6. Default credentials are:
    * Super Admin -  username: admin3@oncloudtime.com, password: 12345678
    * Admin - username: ATS003@oncloudtime.com, password: 12345678
    * Regular user - username: ATS0043@oncloudtime.com, password: 12345678

------------------------------------------------------------------------------------------------------------------------

## 0.5.0 -

1. Accounts:

   - Creating an account requires specifying information for the account admin account which is created 
   - Each account has its own folder in the data directory with the name of the account 
   - Each account contains a shared directory where templates and other files can be copied by users with the admin role 
   - The admin role has the ability to reset the data in the account, however this is not supported for the super_admin role
   - An admin user can configure the dashboard, current and historical activity urls for their account. The iframe parts of the urls are added automatially

2. User and security :


shell/updatetask.sh
#!/bin/bash

set -e

script_full_path=$(dirname "$0")
source "$script_full_path/".aws_config
# $1 - rdbms_user
# $2 - rdbms_password
# $3- rdbms_dbname
# $4 - rdbms_host
# $5 - user
# $6 - project
# $7 - startdate
# $8 - enddate
# $9 - maxiumumhoursperday
# ${10} - maxiumumhoursperweek
# ${11} - saturdayworkallowed
# ${12} - sundayworkallowed
# ${13} - publicholiday work allowed
# ${14} - regular rate
# ${15} - overtime rate
# ${16} - approver name
# ${17} - approver email
# ${18} - auto increment id
# ${19} - uuid
# ${20} - s3 region
# ${21} - s3 bucket
# ${22} - s3 presign duration
# ${23) - logged in username
# ${25} - accountid
# ${26} - absolute path to log file
# ${27} - path to s3 folder to upload log file
# ${28} - log file name
# ${29} - ID of user creating the project

echo "uuid " ${13}

mysql -h $1 --user=$2 --password=$3 $4 -v -v <<EOF
UPDATE project_assignment SET \`startdate\` = "$7", \`enddate\` = "$8", \`maximumhoursperday\` = "$9",  \`maximumhoursperweek\` = "${10}",  \`saturdayworkallowed\` = "${11}",\`sundayworkallowed\` = "${12}",\`publicholidayworkallowed\` = "${13}", \`regularrate\`="${14}",\`overtimerate\` = "${15}",\`approvername\` = "${16}", \`approveremail\` = "${17}" WHERE \`id\` = ${18};
EOF
echo 0